src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -423,17 +423,35 @@ const WorkflowView = _.flow(
   async getValidation() {
     const { namespace, name, workflowNamespace, workflowName, signal } = this.props
 
+    this.setState({ snapshotReferenceError: undefined })
     try {
       return await Ajax(signal).Workspaces.workspace(namespace, name).methodConfig(workflowNamespace, workflowName).validate()
     } catch (e) {
       if (e.status === 404) {
-        return false
+        const errmsg = await e.text()
+        // distinguish between snapshot-reference-not-found and workflow-not-found
+        if (errmsg?.includes('Reference name') && errmsg?.includes('does not exist in workspace')) {
+          this.setState(_.set(['snapshotReferenceError', errmsg]))
+          return true
+        } else {
+          return false
+        }
       } else {
         throw e
       }
     }
   }
 
+  async safeGetSnapshotEntityMetadata(googleProject, modifiedConfig) {
```

Comment:
Not sure if `safe` fits well necessarily here. How about

```suggestion
  async maybeGetSnapshotEntityMetadata(googleProject, modifiedConfig) {
```
?